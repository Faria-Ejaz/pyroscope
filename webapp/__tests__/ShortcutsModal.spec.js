import React from "react";
import { configure, mount } from "enzyme";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import Adapter from "enzyme-adapter-react-16";
import { ShortcutProvider } from "react-keybind";
import renderer from "react-test-renderer";

import ShortcutsModal from "../javascript/components/ShortcutsModal";
import Sidebar from "../javascript/components/Sidebar";

const mockStore = configureMockStore();
configure({ adapter: new Adapter() });

const store = mockStore({});

describe("ShortcutsModal", () => {
  it("render correctly ShortcutsModal component", () => {
    const ShortcutsModalComponent = renderer
      .create(
        <Provider store={store}>
          <ShortcutProvider>
            <Sidebar />
          </ShortcutProvider>
        </Provider>
      )
      .toJSON();
    expect(ShortcutsModalComponent).toMatchSnapshot();
  });
  it("When shortcuts are pressed, a shortcuts modal should appears", () => {
    const wrapper = mount(
      <Provider store={store}>
        <ShortcutProvider>
          <Sidebar />
        </ShortcutProvider>
      </Provider>
    );
    wrapper.find("button").at(1).simulate("click");
    expect(wrapper.find(ShortcutsModal).length).toBe(1);
  });
});
