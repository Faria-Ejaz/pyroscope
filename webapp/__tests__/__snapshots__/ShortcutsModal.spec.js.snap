// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortcutsModal render correctly ShortcutsModal component 1`] = `
<div
  className="sidebar"
>
  <span
    className="logo active"
    onClick={[Function]}
  />
  <div
    className="sidebar-item"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-columns fa-w-16 "
        data-icon="columns"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V80c0-26.51-21.49-48-48-48zM224 416H64V160h160v256zm224 0H288V160h160v256z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <div
      className="sidebar-tooltip-wrapper"
    >
      <span
        className="sidebar-tooltip"
      >
        Comparison View - Coming Soon
      </span>
    </div>
  </div>
  <div
    className="sidebar-space-filler"
  />
  <div
    className="sidebar-item"
  >
    <a
      href="https://pyroscope.io/docs"
      rel="noreferrer"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-file-alt fa-w-12 "
        data-icon="file-alt"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm64 236c0 6.6-5.4 12-12 12H108c-6.6 0-12-5.4-12-12v-8c0-6.6 5.4-12 12-12h168c6.6 0 12 5.4 12 12v8zm0-64c0 6.6-5.4 12-12 12H108c-6.6 0-12-5.4-12-12v-8c0-6.6 5.4-12 12-12h168c6.6 0 12 5.4 12 12v8zm0-72v8c0 6.6-5.4 12-12 12H108c-6.6 0-12-5.4-12-12v-8c0-6.6 5.4-12 12-12h168c6.6 0 12 5.4 12 12zm96-114.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
    <div
      className="sidebar-tooltip-wrapper"
    >
      <span
        className="sidebar-tooltip"
      >
        Docs
      </span>
    </div>
  </div>
  <div
    className="sidebar-item"
  >
    <a
      href="https://pyroscope.io/slack"
      rel="noreferrer"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-slack fa-w-14 "
        data-icon="slack"
        data-prefix="fab"
        fill="none"
        focusable="false"
        role="img"
        viewBox="0 0 17 17"
        xmlns="http://www.w3.org/2000/svg"
      >
        <script
          xmlns=""
        />
        <path
          d="M6.233 0a1.7 1.7 0 000 3.4h1.7V1.7a1.7 1.7 0 00-1.7-1.7zm0 4.533H1.7a1.7 1.7 0 000 3.4h4.534a1.7 1.7 0 100-3.4"
          fill="#36C5F0"
        />
        <path
          d="M17 6.233a1.7 1.7 0 10-3.4 0v1.7h1.7a1.7 1.7 0 001.7-1.7zm-4.533.001V1.7a1.7 1.7 0 10-3.4 0v4.533a1.7 1.7 0 003.4 0"
          fill="#2EB67D"
        />
        <path
          d="M10.766 17a1.7 1.7 0 100-3.4h-1.7v1.7a1.7 1.7 0 001.7 1.7zm0-4.533H15.3a1.7 1.7 0 100-3.4h-4.534a1.7 1.7 0 100 3.4z"
          fill="#ECB22E"
        />
        <path
          d="M0 10.766a1.7 1.7 0 103.4 0v-1.7H1.7a1.7 1.7 0 00-1.7 1.7zm4.533 0V15.3a1.7 1.7 0 103.4 0v-4.534a1.7 1.7 0 10-3.4 0z"
          fill="#E01E5A"
        />
      </svg>
    </a>
    <div
      className="sidebar-tooltip-wrapper"
    >
      <span
        className="sidebar-tooltip"
      >
        Slack
      </span>
    </div>
  </div>
  <div
    className="sidebar-item"
  >
    <a
      href="https://github.com/pyroscope-io/pyroscope"
      rel="noreferrer"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-github fa-w-16 "
        data-icon="github"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 496 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
    <div
      className="sidebar-tooltip-wrapper"
    >
      <span
        className="sidebar-tooltip"
      >
        GitHub
      </span>
    </div>
  </div>
  <div
    className="sidebar-item"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-keyboard fa-w-18 "
        data-icon="keyboard"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M528 448H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h480c26.51 0 48 21.49 48 48v288c0 26.51-21.49 48-48 48zM128 180v-40c0-6.627-5.373-12-12-12H76c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm-336 96v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm-336 96v-40c0-6.627-5.373-12-12-12H76c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm288 0v-40c0-6.627-5.373-12-12-12H172c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h232c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <div
      className="sidebar-tooltip-wrapper"
    >
      <span
        className="sidebar-tooltip"
      >
        Keyboard Shortcuts
      </span>
    </div>
  </div>
</div>
`;
